From 88afded6f0eed9aa4027bffac56f1d43a7905cf6 Mon Sep 17 00:00:00 2001
From: David Sniatkiwicz <davidsn@marvell.com>
Date: Mon, 11 Jun 2018 17:17:10 +0300
Subject: [PATCH 0652/1051] arm64: a8kp: align stream-ids to firmware

This patch aligns all the SIDs to the firmware.

Change-Id: I1388aa14aa6022a769b1570c777f5c45f62a6ceb
Signed-off-by: David Sniatkiwicz <davidsn@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap810.dtsi |  3 ++-
 include/dt-bindings/soc/ap810-stream-id.h     | 18 +++++++++++++-----
 2 files changed, 15 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap810.dtsi b/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
index 3ae4d0302cf3..94d19e7e0f8f 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
@@ -74,7 +74,8 @@
 };
 
 /* GIC ITS in this AP is MSI parent for XOR engines in CPs south-bridges connected to this AP  */
-#define XOR_MSI_PARENT(XOR_NUM) <&AP810_LABEL(gic_its) MVEBU_STREAM_ID_CP_XORx(XOR_NUM)>
+#define XOR_MSI_PARENT(XOR_NUM) <&AP810_LABEL(gic_its) \
+				MVEBU_STREAM_ID_CPx_XORx(CP110_NUM, XOR_NUM)>
 
 / {
 	AP810_NAME {
diff --git a/include/dt-bindings/soc/ap810-stream-id.h b/include/dt-bindings/soc/ap810-stream-id.h
index ef347558d760..9d32ba46d3e9 100644
--- a/include/dt-bindings/soc/ap810-stream-id.h
+++ b/include/dt-bindings/soc/ap810-stream-id.h
@@ -12,12 +12,20 @@
 						(ap * MVEBU_STREAM_ID_AP_MAX) + x)
 
 /* CP stream ID base */
-#define MVEBU_STREAM_ID_BASE			0x80
+#define MVEBU_STREAM_ID_BASE			0x40
+
+/* CP max stream IDs */
+#define MVEBU_MAX_STREAM_ID_PER_CP		0x10
 
 /* CP XOR0 stream ID offset */
-#define MVEBU_STREAM_ID_CP_XOR0_OFFSET		0x9
+#define MVEBU_STREAM_ID_CP_XOR0_OFFSET		0x2
+
+/* CP XOR0 stream ID: 0x42 */
+#define MVEBU_STREAM_ID_CP0_XOR0		(MVEBU_STREAM_ID_BASE + \
+						MVEBU_STREAM_ID_CP_XOR0_OFFSET)
+
+#define MVEBU_STREAM_ID_CPx_XORx(cp_num, xor_num)	\
+	(MVEBU_STREAM_ID_CP0_XOR0 + \
+	((cp_num  % 4) * MVEBU_MAX_STREAM_ID_PER_CP) + xor_num)
 
-/* CP XOR0 stream ID: 0x89 */
-#define MVEBU_STREAM_ID_CP_XOR0			(MVEBU_STREAM_ID_BASE + MVEBU_STREAM_ID_CP_XOR0_OFFSET)
 
-#define MVEBU_STREAM_ID_CP_XORx(xor_num)	(MVEBU_STREAM_ID_CP_XOR0 + xor_num)
-- 
2.17.1

